bullet/src/EntityManagementFeatures.cc

```diff
@@ -216,36 +281,99 @@ Identity EntityManagementFeatures::GetEngineOfWorld(const Identity &) const
 std::size_t EntityManagementFeatures::GetModelCount(
     const Identity &) const
 {
-  return 0;
+  return this->models.size();
 }
 
 Identity EntityManagementFeatures::GetModel(
-    const Identity &, std::size_t) const
+    const Identity &, std::size_t _modelIndex) const
 {
-  return this->GenerateIdentity(0);
+  std::size_t index = 0;
+  std::size_t modelEntity = 0;
+  ModelInfoPtr modelPtr = nullptr;
+
+  for (auto itModel = this->models.begin();
+       itModel != this->models.end();
+       ++itModel)
+  {
+    if (index++ == _modelIndex)
+    {
+      modelPtr = itModel->second;
+      modelEntity = itModel->first;
+      break;
+    }
+  }
+
+  return this->GenerateIdentity(modelEntity, modelPtr);
 }
 
 Identity EntityManagementFeatures::GetModel(
-    const Identity &, const std::string &) const
+    const Identity &, const std::string &_modelName) const
 {
-  return this->GenerateIdentity(0);
+  std::size_t modelEntity = 0;
+  ModelInfoPtr modelPtr = nullptr;
+  for (auto itModel = this->models.begin();
+       itModel != this->models.end();
+       ++itModel)
+  {
+    if (itModel->second->name == _modelName)
+    {
+      modelPtr = itModel->second;
+      modelEntity = itModel->first;
+      break;
+    }
+  }
+
+  return this->GenerateIdentity(modelEntity, modelPtr);
 }
 
 const std::string &EntityManagementFeatures::GetModelName(
-    const Identity &) const
+    const Identity &_modelID) const
 {
-  static const std::string modelName = "bulletModel";
+  // Check if the model exists
+  if (this->models.find(_modelID.id) == this->models.end())
+  {
+    static const std::string modelName = "";
+    return modelName;
+  }
+
+  static const std::string modelName = this->models.at(_modelID)->name;
   return modelName;
 }
 
-std::size_t EntityManagementFeatures::GetModelIndex(const Identity &) const
+std::size_t EntityManagementFeatures::GetModelIndex(
+  const Identity &_modelId) const
 {
+  if (this->models.find(_modelId.id) == this->models.end())
+  {
+    return 0;
+  }
+
+  std::size_t index = 0;
+  ModelInfoPtr modelPtr = nullptr;
+
+  for (const auto& [key, value] : this->models)
+  {
+    if (key == _modelId.id)
+    {
+      return index;
+    }
+    index++;
+  }
+
   return 0;
 }
 
-Identity EntityManagementFeatures::GetWorldOfModel(const Identity &) const
+Identity EntityManagementFeatures::GetWorldOfModel(
+  const Identity &_modelId) const
 {
-  return this->GenerateIdentity(0);
+  if (this->models.find(_modelId.id) == this->models.end())
```

Comment:
Could also use a bit shorter, since you aren't doing anything with the result:

```suggestion
  if (!this->models.count(_modelId.id))
```